docs/compiler.md

```diff
@@ -2,66 +2,37 @@
 
 The neo-go compiler compiles Go programs to bytecode that the NEO virtual machine can understand.
 
-## Currently supported
-
-### Go internals 
-- type checking
-- multiple assignments
-- global variables
-- types int, string, byte and booleans
-- struct types + method receives
-- functions
-- composite literals `[]int, []string, []byte`
-- basic if statements
-- binary expressions
-- return statements
-- for loops
-- imports 
-
-### Go builtins
-- len
-- append
-
-### VM API (interop layer)
+## Language compatibility
+
+The compiler is mostly compatible with regular Go language specification, but
+there are some important deviations that you need to be aware of that make it
+a dialect of Go rather than a complete port of the language:
+ * `make()` ane `new()` are not supported, most of the time you can substitute
+   them with composite literals
+ * there is no real distinction between different integer types, all of them
+   work as big.Int in Go with a limit of 256 bit in width, so you can use
+   `int` for just about anything. This is the way integers work in Neo VM and
+   adding proper Go types emulation is considered to be too costly.
+ * goroutines, channels and garbage collection are not supported and will
+   never be because emulating that aspects of Go runtime on top of Neo VM is
+   close to impossible
+ * even though `panic()` is supported, `recover()` is not, `panic` shuts the
+   VM down
+ * lambdas are not supported
+ * global variables can't be changed in functions (#638)
+ * it's not possible to rename imported interop packages, they won't work this
+   way (#397, #913)
+
```

Comment:
Are nested selectors going to be supported? From my experience I often want to use them when cycling slice of structures, but can't. If not, then it is worth mention it there.
```go
for i := 0; i < len(arr); i++ {
    foo(arr[i].bar)
}
```

Reply:
Please file an issue for it (it should be fixed eventually) and I'll add it to this list.

Reply:
#957 

Also don't forget that range loops with value variable are also not supported.

Reply:
OK, #958.